### PR TITLE
Add Typescript support

### PR DESCRIPTION
**Pull request by ipfans**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)

None.

##### Description of change

Add Typescript support.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,9 +5,11 @@
   "eggPlugin": {
     "name": "redis"
   },
+  "types": "./index.d.ts",
```
Comment (whxaxes) [inline package.json:8]: Also need to add `@types/ioredis` in dependencies?

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { Redis, RedisOptions, ClusterOptions } from "ioredis";
+
+declare module 'egg' {
+  interface Application {
+    redis: Redis;
+  }
+
+  interface EggAppConfig {
+    redis: RedisOptions | ClusterOptions;
```
Comment (whxaxes) [inline index.d.ts:9]: It seems incorrect with this. see https://github.com/eggjs/egg-redis/blob/master/config/config.default.js#L3

**Review by whxaxes — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (ipfans) [reply package.json:8]: `ioredis` created `index.d.ts` for use it directly.

Comment (whxaxes) [reply package.json:8]: Are you sure ? https://github.com/luin/ioredis But I can't find any types file in this repo

**Comment by whxaxes** [comment]
Is there any progress?

**Comment by ipfans** [comment]
@whxaxes yep, I may submit new commit in days.

**Comment by songyazhao** [comment]
迫不及待 😉

**Comment by Micjoyce** [comment]
是否有新的进展呢？

**Comment by ipfans** [comment]
@Micjoyce 不好意思，之后住院了，刚出院，尽快更新一下

**Comment by Micjoyce** [comment]
@ipfans 祝早日康复

**Comment by xiaocongWang** [comment]
@ipfans   祝早日康复，请问下最近有更新的计划吗~

**Comment by ipfans** [comment]
I will re-submit PR in recently, so close this one.

**Pull request closed without merging**